### PR TITLE
Two sets of localtest fixes for July 2022

### LINKED ISSUE #13323: SerializationTests::Actions started failing recently
### Windows Terminal version

Commit 848314ef17419430153561016d1fabe912338844

### Windows build number

10.0.19044.1706

### Other Software

_No response_

### Steps to reproduce

1. Build a recent commit of OpenConsole (I think anything after 799b5d4add799ac2fbf5c855b774cd7dcb957697 should do).
2. Run the unit tests.

### Expected Behavior

All the tests should pass.

### Actual Behavior

There is a failure in `SettingsModelLocalTests::SerializationTests::Actions` when attempting to serialize the pair of `findMatch` actions. It seems that the commands are being reordered.

<details>
  <summary>Click to see error</summary>
  
```
Error: Verify: AreEqual(toString(json), toString(result)) - Values ([
        {
                "command" :
                {
                        "action" : "findMatch",
                        "direction" : "next"
                },
                "keys" : "ctrl+shift+s"
        },
        {
                "command" :
                {
                        "action" : "findMatch",
                        "direction" : "prev"
                },
                "keys" : "ctrl+shift+r"
        }
], [
        {
                "command" :
                {
                        "action" : "findMatch",
                        "direction" : "prev"
                },
                "keys" : "ctrl+shift+r"
        },
        {
                "command" :
                {
                        "action" : "findMatch",
                        "direction" : "next"
                },
                "keys" : "ctrl+shift+s"
        }
])
```
</details>

I'm not sure if this is actually an indication of a bug - is the order of the commands expected to be retained? But at the very least I'd expect the test to be corrected if this is just an altered hash map order or something of that sort.

If it helps, I think the first time the test stopped working was after commit 799b5d4add799ac2fbf5c855b774cd7dcb957697.

### LINKED ISSUE #13474: Lots of local unit tests are failing again
### Windows Terminal version

Commit bbc570d107b6b67b68454355e04724ecea9c9796

### Windows build number

10.0.19044.1706

### Other Software

_No response_

### Steps to reproduce

1. Checkout the latest code.
2. Build.
3. Run the unit tests.

### Expected Behavior

All tests should pass.

### Actual Behavior

I'm seeing 38 failures. I couldn't capture the full list, but this is some of them:

    SettingsModelLocalTests::DeserializationTests::ValidateDefaultProfileExists [Failed]
    SettingsModelLocalTests::DeserializationTests::ValidateDuplicateProfiles [Failed]
    SettingsModelLocalTests::DeserializationTests::ValidateManyWarnings [Failed]
    SettingsModelLocalTests::DeserializationTests::TestInvalidColorSchemeName [Failed]
    SettingsModelLocalTests::DeserializationTests::ValidateKeybindingsWarnings [Failed]
    SettingsModelLocalTests::DeserializationTests::ValidateColorSchemeInCommands [Failed]
    SettingsModelLocalTests::DeserializationTests::ValidateExecuteCommandlineWarning [Failed]
    SettingsModelLocalTests::DeserializationTests::TestNestedCommandWithoutName [Failed]
    SettingsModelLocalTests::DeserializationTests::TestNestedCommandWithBadSubCommands [Failed]
    SettingsModelLocalTests::SerializationTests::Actions [Failed]
    SettingsModelLocalTests::SerializationTests::CascadiaSettings [Failed]
    SettingsModelLocalTests::TerminalSettingsTests::MakeSettingsForDefaultProfileThatDoesntExist [Failed]
    SettingsModelLocalTests::ThemeTests::ParseSimpleTheme [Failed]
    SettingsModelLocalTests::ThemeTests::ParseNoWindowTheme [Failed]
    SettingsModelLocalTests::ThemeTests::ParseNullWindowTheme [Failed]

    TerminalAppLocalTests::SettingsTests::TestIterateCommands [Failed]
    TerminalAppLocalTests::SettingsTests::TestIterateOnGeneratedNamedCommands [Failed]
    TerminalAppLocalTests::SettingsTests::TestIterateOnBadJson [Failed]
    TerminalAppLocalTests::SettingsTests::TestNestedCommands [Failed]
    TerminalAppLocalTests::SettingsTests::TestNestedInNestedCommand [Failed]
    TerminalAppLocalTests::SettingsTests::TestNestedInIterableCommand [Failed]
    TerminalAppLocalTests::SettingsTests::TestIterableInNestedCommand [Failed]
    TerminalAppLocalTests::SettingsTests::TestMixedNestedAndIterableCommand [Failed]
    TerminalAppLocalTests::TabTests::TryInitializePage [Failed]
    TerminalAppLocalTests::TabTests::TryDuplicateBadTab [Failed]
    TerminalAppLocalTests::TabTests::TryDuplicateBadPane [Failed]
    TerminalAppLocalTests::TabTests::TryZoomPane [Failed]
    TerminalAppLocalTests::TabTests::MoveFocusFromZoomedPane [Failed]
    TerminalAppLocalTests::TabTests::CloseZoomedPane [Failed]
    TerminalAppLocalTests::TabTests::SwapPanes [Failed]
    TerminalAppLocalTests::TabTests::NextMRUTab [Failed]
    TerminalAppLocalTests::TabTests::VerifyCommandPaletteTabSwitcherOrder [Failed]
    TerminalAppLocalTests::TabTests::TestWindowRenameSuccessful [Failed]
    TerminalAppLocalTests::TabTests::TestWindowRenameFailure [Failed]
    TerminalAppLocalTests::TabTests::TestPreviewCommitScheme [Failed]

### PR DESCRIPTION
**Pull request by zadjii-msft**
Cleans up a couple local test failures.

* [x] Closes #13474: So, I clearly hadn't ran the local tests at the end of the themes PR. We needed a sensible fallback to SOME theme, even if there wasn't one provided in the user json. This is only really hit in the tests (that don't also include `defaults.json`.
* [x] Closes #13323: Meh, the ordering of the keys in this test doesn't matter. Ordering is a map implementation detail. This is fine.
* [x] Ran tests locally

### CODE REVIEW

File: src/cascadia/LocalTests_SettingsModel/SerializationTests.cpp [hunk 1]
```diff
@@ -253,9 +253,14 @@ namespace SettingsModelLocalTests
                                                 { "command": { "action": "adjustFontSize", "delta": 1 }, "keys": "ctrl+c" },
                                                 { "command": { "action": "adjustFontSize", "delta": 1 }, "keys": "ctrl+d" }
                                             ])" };
+        // GH#13323 - these can be fragile. In the past, the order these get
+        // re-serialized as has been not entirely stable. We don't really care
+        // about the order they get re-serialized in, but the tests aren't
+        // clever enough to compare the structure, only the literal string
+        // itself. Feel free to change as needed.
```
Comment (DHowett) [inline src/cascadia/LocalTests_SettingsModel/SerializationTests.cpp:260]: UGH.

File: src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp [hunk 2]
```diff
@@ -1156,6 +1156,18 @@ void CascadiaSettings::ExportFile(winrt::hstring path, winrt::hstring content)
 
 void CascadiaSettings::_validateThemeExists()
 {
+    if (_globals->Themes().Size() == 0)
+    {
+        // GH #13474
+        // We didn't even load the default themes. This should only be possible
+        // in the tests that don't even pass in defaults.json. No matter. Create
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp:1163]: Comments that say "this never happens" _do not age well_; reconsider!

(also: linking it to a github ID isn't meaningful in this case. we just should always make sure there's a theme, regardless of a github issue. I generally hold that if we take this to its logical extreme, every line of code will have its own github issue . . . and then it's no better than `git blame` but in many ways worse)

File: src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp [hunk 3]
```diff
@@ -1156,6 +1156,18 @@ void CascadiaSettings::ExportFile(winrt::hstring path, winrt::hstring content)
 
 void CascadiaSettings::_validateThemeExists()
 {
+    if (_globals->Themes().Size() == 0)
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp:1159]: - If this is only hit in the tests, shouldn't we be adding the "system" theme in the tests?
- In a (possibly distant) world where we have multiple `GlobalAppSettings` in the inheritance structure, I could see some of them not having a theme set. Is that going to be a problem here where we add an empty "system" theme?

Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp:1159]: The empty system theme _IS_ the system theme (that's actually how it works!)

I think it's important to have in the validation, honestly. defaults.json is a crutch, right?

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp:1159]: I'm rewording the comment here. 

I think it's fine to leave this as a validation step. In reality, we'll always have a `themes` in `defaults.json`, but oftentimes the tests just don't even bother with a `defaults`, for simplicity. 

The alternative would be to make the themes in `defaults.json` actually hardcoded instead, but that seemed more fragile. Seemed like it was a good idea to have the themes actually listed there (esp. for the `terminalBackground` treatment we're giving the tabs in #13554)

**Review by carlos-zamora — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Pull request merged**